src/server/project.ts

```diff
@@ -1006,6 +1051,61 @@ namespace ts.server {
             return !!this.missingFilesMap && this.missingFilesMap.has(path);
         }
 
+        /* @internal */
+        addGeneratedFileWatch(generatedFile: string, sourceFile: string) {
+            if (this.compilerOptions.outFile || this.compilerOptions.out) {

```

Comment:
I'm not really understanding the connection between `outFile`/`out` and the single watcher vs. multi watcher scenario. Can you give me a little background?

Reply:
If options specify `outFile` or `out` there is going to be single generated file for all source files in the program. Otherwise 1:1 mapping between source and generated files

Reply:
I was definitely reading this as `outFile`/`outDir` in my head, and was confused how it could apply to `outDir`—and the answer is it doesn’t. Makes sense now, thanks 🤦‍♂